v3/newrelic/segments.go

```diff
@@ -337,6 +340,36 @@ func StartExternalSegment(txn *Transaction, request *http.Request) *ExternalSegm
 	return s
 }
 
+func StartExternalSegmentFastHTTP(txn *Transaction, ctx *fasthttp.RequestCtx) *ExternalSegment {
+	if nil == txn {
+		txn = transactionFromRequestContextFastHTTP(ctx)
+	}
+	request := &http.Request{}
+
+	fasthttpadaptor.ConvertRequest(ctx, request, true)
+	s := &ExternalSegment{
+		StartTime: txn.StartSegmentNow(),
+		Request:   request,
+	}
+	if IsSecurityAgentPresent() {
+		s.secureAgentEvent = secureAgent.SendEvent("OUTBOUND", request)
+	}
+
+	if request != nil && request.Header != nil {
```

Comment:
`s.outboundHeaders` and `DistributedTraceHeaders` should be added to the original request context before executing the request.

Reply:
Hi @aayush-ap

Looking at this, the StartExternalSegmentFastHTTP function is identical to the StartExternalSegment function. The only difference being is that we convert to a http request. Im a bit confused at this comment because we are adding `s.outboundHeaders` and `DistributedTraceHeaders` before returning the request context and executing the request.


 Unless you mean calling `s.secureAgentEvent = secureAgent.SendEvent("OUTBOUND", request)` *after* we add the two headers?

cc @nr-swilloughby

Reply:
Yes,  you are converting `ctx` fasthttp.RequestCtx to http request and then adding all the required headers.
But that does not reflect the original request object that is used to call an outbound request.

Reply:
Please check the following fix for outbound request

https://github.com/k2io/go-agent-fork/commit/15a72c8992aa1767f7b2596ed84d07307a1d23d3